src/test/java/name/valery1707/kaitai/KaitaiUtilsTest.java

```diff
@@ -112,9 +93,279 @@ public void testScanFiles() throws KaitaiException {
 		;
 	}
 
+	@Test
+	public void testCreateTempDirectory() throws KaitaiException {
+		Path temp = createTempDirectory("temp-");
+		assertThat(temp)
+			.exists()
+			.isDirectory()
+			.isReadable()
+		;
+		assertThat(temp.toFile().listFiles()).isNotNull().isEmpty();
+	}
+
+	private void testDelete(Path path) throws KaitaiException {
+		assertThat(path).exists();
+		delete(path);
+		assertThat(path).doesNotExist();
+	}
+
+	private void testDelete_root(FileSystem fs) {
+		try {
+			testDelete(fs.getRootDirectories().iterator().next());
+			fail("Must generate exception because root can't be deleted");
+		} catch (KaitaiException e) {
+			assertThat(e).hasMessageStartingWith("Fail to delete: ");
+			assertThat(e.getCause()).isNotNull().hasMessageContaining("can not delete root");
+		}
+	}
+
+	@Test
+	public void testDelete_root_Linux() throws IOException {
+		FileSystem fs = MemoryFileSystemBuilder.newLinux().build();
+		testDelete_root(fs);
+	}
+
+	@Test
+	public void testDelete_root_Windows() throws IOException {
+		FileSystem fs = MemoryFileSystemBuilder.newWindows().build();
+		testDelete_root(fs);
+	}
+
+	@Test
+	public void testDelete_file_Linux() throws IOException, KaitaiException {
+		FileSystem fs = MemoryFileSystemBuilder.newLinux().build();
+		testDelete(Files.createFile(fs.getPath(".", "temp.file")));
+	}
+
+	@Test
+	public void testDelete_file_Windows() throws IOException, KaitaiException {
+		FileSystem fs = MemoryFileSystemBuilder.newWindows().build();
+		testDelete(Files.createFile(fs.getPath(".", "temp.file")));
+	}
+
+	private void testDelete_fileLocked(FileSystem fs) throws IOException {
+		Path locked = Files.createFile(fs.getPath(".", "temp.file"));
+		try (InputStream ignored = Files.newInputStream(locked)) {
+			testDelete(locked);
+			fail("Must generate exception because file is locked");
+		} catch (KaitaiException e) {
+			assertThat(e).hasMessageStartingWith("Fail to delete:").hasMessageEndingWith(locked.getFileName().toString());
+			assertThat(e.getCause()).hasMessageContaining("file still open");
+		}
+	}
+
+	@Test
+	public void testDelete_fileLocked_Linux() throws IOException {
+		FileSystem fs = MemoryFileSystemBuilder.newLinux().build();
+		testDelete_fileLocked(fs);
+	}
+
+	@Test
+	public void testDelete_fileLocked_Windows() throws IOException {
+		FileSystem fs = MemoryFileSystemBuilder.newWindows().build();
+		testDelete_fileLocked(fs);
+	}
+
+	private Path createPathTree(Path root, Map<String[], Character> content) throws IOException {
+		for (String[] paths : content.keySet()) {
+			Path path = root;
+			for (String part : paths) {
+				path = path.resolve(part);
+			}
+			switch (content.get(paths)) {
```

Comment:
\[FindBugs\] INFO: WMI: name\.valery1707\.kaitai\.KaitaiUtilsTest\.createPathTree\(Path, Map\) makes inefficient use of keySet iterator instead of entrySet iterator